### PR TITLE
Support paths with datetime x-axis (nschloe#328)

### LINKED ISSUE #328: Conversion of fill_between to \path fails when using datetime axes
The following MWE creates a simple plot with a line and a shaded area between two other lines using `pyplot.fill_between`.

```python
import matplotlib.pyplot as plt
import matplotlib.dates as mdates
import tikzplotlib
from datetime import datetime, timedelta

times = datetime.now(), datetime.now() + timedelta(seconds=3600)

line = [2, 2]
upper = [3, 4]
lower = [1, 0]

plt.plot(times, line)
plt.fill_between(times, lower, upper, edgecolor='darkgray', facecolor='darkgray')

ax = plt.gca()

ax.fmt_xdata = mdates.DateFormatter('%d %b %Y %H:%M:%S')

tikzplotlib.save('fig.tex')

plt.show()
```

The Tikz code saved by tikzplotlib to file `fig.tex` looks like this:

```LaTeX
% This file was created by tikzplotlib v0.8.2.
\begin{tikzpicture}

\definecolor{color0}{rgb}{0.12156862745098,0.466666666666667,0.705882352941177}

\begin{axis}[
date coordinates in=x,
tick align=outside,
tick pos=left,
x grid style={white!69.01960784313725!black},
xmin=2019-08-25 01:02, xmax=2019-08-25 02:08,
xtick style={color=black},
y grid style={white!69.01960784313725!black},
ymin=-0.2, ymax=4.2,
ytick style={color=black}
]
\path [draw=white!66.27450980392156!black, fill=white!66.27450980392156!black]
(axis cs:737296.045298025,3) % Incorrect: Float values as x-coordinates
--(axis cs:737296.045298025,1)
--(axis cs:737296.086964692,0)
--(axis cs:737296.086964692,4)
--(axis cs:737296.086964692,4)
--(axis cs:737296.045298025,3)
--cycle;

\addplot [semithick, color0]
table [header=false,col sep=comma] {
2019-08-25 01:05,2 % Correct: Datetime values as x-coordinates
2019-08-25 02:05,2
};
\end{axis}

\end{tikzpicture}
```

Note the inconsistency in the type of the x-coordinates in the `\path` command vs the `\addplot` command. Including the figure in a LaTeX document `main.tex`...

```LaTeX
\documentclass[11pt,a4paper]{article}

\usepackage[utf8]{inputenc}
\usepackage{pgfplots}
\pgfplotsset{compat=newest}
\usepgfplotslibrary{groupplots}
\usepgfplotslibrary{dateplot}

\begin{document}

\input{fig.tex}

\end{document}
```

...confirms that this won't work:

```bash
$ pdflatex main.tex && evince main.pdf

[Output truncated]

(./fig.tex
! Argument of \pgfcalendar@datetojulian has an extra }.
<inserted text> 
                \par 
l.31 \end{axis}

? 
```

Commenting out the `plt.plot(times, line)` call in the Python script leads to a non-failing run of `pdflatex`. However, by inspecting the graph in the resulting PDF we can see that this is due to pgfplots falling back to a float axis in the absence of any datetime coordinate values.

### PR DESCRIPTION
**Pull request by harenbrs**
Fixes #328.

### CODE REVIEW

File: tikzplotlib/_path.py [hunk 1]
```diff
@@ -22,8 +23,10 @@ def draw_path(data, path, draw_options=None, simplify=None):
     ):
         return data, "", None, False
 
+    x_is_date = isinstance(data["current mpl axes obj"].xaxis.converter, DateConverter)
```
Comment (harenbrs) [inline tikzplotlib/_path.py]: If there's a better way to detect the presence of a datetime axis, please edit accordingly

**Comment by nschloe** [comment]
I'd be great to have a test for it, too.

**Comment by harenbrs** [comment]
I've added one now

**Comment by nschloe** [comment]
LGTM, thanks!

**Pull request merged**